### DISCUSSION TITLE
http://example.lgsl.com/?s=3 to http://example.lgsl.com/ipserver:port

### DISCUSSION THREAD

**Discussion by Kent-Romania** [tltneon/lgsl#107] [Q&A] [answered]
Hello. can someone help me modify the **lgsl_class.php** file part:
   function lgsl_link ($ s = "")
   {
     global $ lgsl_config, $ lgsl_url_path;

     $ index = $ lgsl_config ['direct_index']? "index.php": "";

     switch ($ lgsl_config ['cms'])
     {
       case "e107":
         $ link = $ s? e_PLUGIN_ABS. "lgsl / {$ index}? s = {$ s}": e_PLUGIN_ABS. "lgsl / {$ index}";
       break;

       "joomla" case:
         $ link = $ s? JRoute :: _ ("index.php? Option = com_lgsl & s = {$ s}"): JRoute :: _ ("index.php? Option = com_lgsl");
       break;

       "drupal" case:
         $ link = $ s? url ("LGSL / {$ s}"): url ("LGSL");
       break;

       case "phpnuke":
         $ link = $ s? "modules.php? name = LGSL & s = {$ s}": "modules.php? name = LGSL";
       break;

       default: // "sa"
         $ link = $ s? $ lgsl_url_path. "../{$ index}? s = {$ s}": $ lgsl_url_path. "../{$ index}";
       break;
     }

     return $ link;
   }
================================
This change will bring the result http://example.lgsl.com/**?s=3** to http://example.lgsl.com/**ipserver:port**

**Comment by tltneon** [comment]
Hello @Kent-Romania 

Just modifying lgsl_link() is not enough to implement this thing 😃 

Do you want solution or asking for help with learning LGSL?

**Reply by Kent-Romania** [reply]
Yes, I want a solution, it would help. or better an update in lgsl_master. but it would help me a lot. :)

**Reply by BaxAndrei** [reply]
Nu o vad rostul pentru o asa modificare.. Daca ai fi cautat putin prin cod inceand cu `index.php` ai fi gasit deja ce te intereseaza. O sa vezi ca de acolo se seteaza `$_GET['s']` in `$lgsl_server_id` care este folosit in `fisierul lgsl_details.php` pentru a seta `$server` cu datele returante de `lgsl_query_cached` si le foloseste iar in `index.php`. ID-ul este cel mai simplu si cel mai practic mod de a stoca/identifica serverele.

**Reply by Kent-Romania** [reply]
Merci dar astept si raspunsul lui tltneon 😉

**Reply by BaxAndrei** [reply]
Cum sunt romanii mei :))
Dat fiind ca nu contribui la proiectul asta si nici nu iti dai interesul sa inveti, daca te uitai prin issues ai fi aflat ca nu prea are timp asa mult. Toti care au vrut sa contribuie si sa faca modificari la acest proiect au trimis merge request-uri cu modificarile, nu au asteptat sa pice din cer 😉

**Reply by Kent-Romania** [reply]
Ajuta-ma tu, eu nu prea am timp sincer, din motivul asta cer ajutor.

**Reply by BaxAndrei** [reply]
Din pacate nici eu nu am timp, dar modificarea asta iti zic sincer ca nu isi are neaparat rosul si nu merita timpul, trebuie sa rescrii o mare parte din program, ce are ID-ul? Asa nu apar nici probleme daca schimbi ip/port.

**Reply by tltneon** [reply]
@Kent-Romania 
Try out latest update. I changed link from ?s= to ?ip=&port=

If you using Nginx you need to add something like this to your website.conf
```shell
	location ~ /(([0-9]+)\.([0-9]+)\.([0-9]+)\.([0-9]+)|[a-zA-Z\.]+):([0-9][0-9]+)$ {
		rewrite ^/(.*):(.*)$ /index.php?ip=$1&port=$2 last;
	}
```

And it will be working

**Reply by Kent-Romania** [reply]
Hi, thank you very much for your effort, I did a test, but it still needs something, sorry but it would look better from:
https://test.mortall.ro/?ip=CS.Mortall.RO&port=27015
in **https://test.mortall.ro/CS.Mortall.RO:27015**
I know you don't have time, but maybe it would help more people, including me.
It's not a rush when you have time.
Thanks 👍 <3

**Reply by BaxAndrei** [reply]
Salut, ai incercat sa pui regula de rewrite?
![image](https://user-images.githubusercontent.com/12484505/165467568-448a0e19-2d7d-4276-9076-de2df38f88bf.png)

Incearca sa pui asta in htaccess:
```
RewriteRule ^(([0-9]+)\.([0-9]+)\.([0-9]+)\.([0-9]+)|[a-zA-Z\.]+):([0-9][0-9]+)$ index.php?ip=$1&port=$2
```

**Reply by Kent-Romania** [reply]
> 

Salut,
da am incercat. tot la fel https://test.mortall.ro/?ip=CS.Mortall.RO&port=27015

**Reply by Kent-Romania** [reply]
Howdy. Is it possible to try to work on my wish? I still haven't managed to find the solution to make the link to my liking. this is how it should look **https://test.mortall.ro/cs.mortall.ro:27015** but I can't. help me do you have time?

**Reply by tltneon** [reply]
Hello. Do you using Apache webserver or nginx? @BaxAndrei posted a solution in case of using Apache https://github.com/tltneon/lgsl/discussions/107#discussioncomment-2644537

**Reply by Kent-Romania** [reply]
nginx

**Reply by tltneon** [reply]
It should be something like that:
`
rewrite ^/\?ip=([^&]*)&port=([^&]*) /$1:$2 last;`

**Reply by Kent-Romania** [reply]
Hello, I put this rewrite ^/\?ip=([^&]*)&port=([^&]*) /$1:$2 last; in .htaccess and the result is Error 500!
Internal Server Error

**Reply by tltneon** [reply]
Nginx don't use .htaccess files. Are you sure you using nginx?

**Reply by BaxAndrei** [reply]
> nginx

![image](https://user-images.githubusercontent.com/12484505/184059001-f3844e74-e132-42a1-9ca0-861ad9d8504e.png)

**Reply by Kent-Romania** [reply]
Howdy. finally what I use in .htaccess. which option should I use?

**Reply by BaxAndrei** [reply]
```
RewriteEngine On
RewriteRule ^(([0-9]+)\.([0-9]+)\.([0-9]+)\.([0-9]+)|[a-zA-Z\.]+):([0-9][0-9]+)$ index.php?ip=$1&port=$2
```

**Reply by Kent-Romania** [reply]
the result after adding in .htaccess is https://test.mortall.ro/?ip=PRO.Mortall.RO&port=27018
I think you understood what I want: ex the result is https://test.mortall.ro/pro.mortall.ro:27018
  or as they say from gametraker https://www.gametracker.com/server_info/86.122.123.109:27018/

**Comment by BaxAndrei** [comment, marked as answer]
Incearca cu asta:
```
RewriteEngine On
RewriteRule ^([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+|[a-zA-Z0-9\.]+):([0-9][0-9]+)$ index.php?ip=$1&port=$2
```

DEMO: https://apps.baxandrei.ro/.bax/testrewrite/pro.mortall.ro:27018

Merge si cu `RewriteRule ^([a-zA-Z0-9\.]+):([0-9][0-9]+)$ index.php?ip=$1&port=$2` pt ca nu are rost sa verifici prima data pt un ip 1.2.3.4 cand la urmatoarea verifici orice string din litere si numere

**Reply by Kent-Romania** [reply]
> Incearca cu asta:
> 
> ```
> RewriteEngine On
> RewriteRule ^([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+|[a-zA-Z0-9\.]+):([0-9][0-9]+)$ rewrite.php?ip=$1&port=$2
> ```
> 
> DEMO: https://apps.baxandrei.ro/.bax/testrewrite/pro.mortall.ro:27018
> 
> Merge si cu `RewriteRule ^([a-zA-Z0-9\.]+):([0-9][0-9]+)$ rewrite.php?ip=$1&port=$2` pt ca nu are rost sa verifici prima data pt un ip 1.2.3.4 cand la urmatoarea verifici orice string din litere si numere

Nu merge, la fel

**Reply by BaxAndrei** [reply]
Imposibil, ai si exemplu si merge perfect

**Reply by Kent-Romania** [reply]
Eu zic sa urci lgsl pe host sa faci un test la tine, daca nu iti dau acces la mine sa vezi despre ce e vorba.

**Comment by BaxAndrei** [comment]
`index.php` instead of `rewrite.php`

**Reply by Kent-Romania** [reply]
Perfect works very well. thank you very much for the effort.
